src/main/java/org/apache/solr/benchmarks/BenchmarksMain.java

```diff
@@ -450,15 +508,28 @@ public Map values() {
 		}
 	}
 
-	private static class ErrorListener implements ControlledExecutor.ExecutionListener<String, NamedList<Object>> {
+	private static class ValidationListener implements ControlledExecutor.ExecutionListener<String, SolrBenchQueryResponse> {
 
+		public List<SolrBenchQueryResponse> queryResponses = new Vector<>();
+		
 		@Override
-		public void onExecutionComplete(String typeKey, NamedList<Object> result, long duration) {
+		public void onExecutionComplete(String typeKey, SolrBenchQueryResponse result, long duration) {
+			queryResponses.add(result);
```

Comment:
I have some concerns on memory usage as this appears to store all responses until the benchmark finishes.

I don't think it's a blocker right now, as it should be okay  if the response size is small or if the total amount queries executed are low.  

Perhaps in the future we can consider doing the validation at onExecutionComplete itself? I assume the overall computation is the same - validate each result here vs iterate all responses and validate at the end. It might increase the total duration per task, but per query duration (which is what matters more?) should be the same as far as we disregard the time spent in validation :) ? 

Another minor benefit of doing it as here is we can fail fast if we want. 

Similarly, for the run that generates validation (instead of performing validation), we can as well just keep the doc count here instead of the whole response?

Anyway, not a blocker I think 😊